### DISCUSSION TITLE
Connect to the PostgreSQL server

### DISCUSSION THREAD

**Discussion by TronMetatron** [TransformerOptimus/SuperAGI#189] [Q&A]
I have been working with your Docker containers for superagi-super__postgres-1 and superagi-celery-1. under SuperAGI
I've noticed an issue where the Celery worker fails to connect to the PostgreSQL server, with an error message suggesting that the server may not be accepting TCP/IP connections on port 5432.

Upon investigation, I found that both containers are correctly connected to the superagi_super_network and should be able to communicate with each other. The PostgreSQL server has the IP address 172.18.0.2, and the Celery worker has the IP address 172.18.0.6.

One potential cause for the issue could be that the PostgreSQL server is not ready to accept connections when the Celery worker tries to connect to it. If this is the case, a possible solution could be to add a startup script that waits for the PostgreSQL server to be ready before starting the Celery worker. log 

superagi-celery-1           |   Is the server running on host "super__postgres" (172.18.0.2) and accepting
superagi-celery-1           |   TCP/IP connections on port 5432?
superagi-celery-1           |
superagi-celery-1           | (Background on this error at: https://sqlalche.me/e/14/e3q8)
superagi-super__postgres-1  | 2023-06-06 00:58:16.145 UTC [49] LOG:  checkpoint starting: shutdown immediate
superagi-celery-1           | /usr/local/lib/python3.9/site-packages/celery/platforms.py:840: SecurityWarning: You're running the worker with superuser privileges: this is
superagi-celery-1           | absolutely not recommended!
superagi-celery-1           |
superagi-celery-1           | Please specify a different user using the --uid option.
superagi-celery-1           |
superagi-celery-1           | User information: uid=0 euid=0 gid=0 egid=0
superagi-celery-1           |
superagi-celery-1           |   warnings.warn(SecurityWarning(ROOT_DISCOURAGED.format(
superagi-celery-1           |
superagi-celery-1           |  -------------- celery@e2241117e38b v5.2.7 (dawn-chorus)
superagi-celery-1           | --- ***** -----
superagi-celery-1           | -- ******* ---- Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.31 2023-06-06 00:58:17
superagi-celery-1           | - *** --- * ---
superagi-celery-1           | - ** ---------- [config]
superagi-celery-1           | - ** ---------- .> app:         superagi:0x7fd1a9771040
superagi-celery-1           | - ** ---------- .> transport:   redis://super__redis:6379/0
superagi-celery-1           | - ** ---------- .> results:     redis://super__redis:6379/0
superagi-celery-1           | - *** --- * --- .> concurrency: 10 (prefork)
superagi-celery-1           | -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
superagi-celery-1           | --- ***** -----
superagi-celery-1           |  -------------- [queues]
superagi-celery-1           |                 .> celery           exchange=celery(direct) key=celery
superagi-celery-1           |

**Comment by neelayan7** [comment]
Hi is this issue still persisting?
